gdal/ogr/ogrsf_frmts/ngw/ogrngwlayer.cpp

```diff
@@ -0,0 +1,1116 @@
+                    break;
+                case OFTReal:
+                    poFeature->SetField( iField, oJSONField.ToDouble() );
+                    break;
+                case OFTBinary:
+                    // Not supported.
+                    break;
+                case OFTString:
+                case OFTIntegerList:
+                case OFTInteger64List:
+                case OFTRealList:
+                case OFTStringList:
+                    poFeature->SetField( iField, oJSONField.ToString().c_str() );
+                    break;
+                case OFTDate:
+                case OFTTime:
+                case OFTDateTime:
+                {
+                    int nYear = oJSONField.GetInteger("year");
+                    int nMonth = oJSONField.GetInteger("month");
+                    int nDay = oJSONField.GetInteger("day");
+                    int nHour = oJSONField.GetInteger("hour");
+                    int nMinute = oJSONField.GetInteger("minute");
+                    int nSecond = oJSONField.GetInteger("second");
+                    poFeature->SetField( iField, nYear, nMonth, nDay, nHour,
+                        nMinute, float(nSecond) );
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+    }
+    OGRGeometry *poGeometry = nullptr;
+    OGRGeometryFactory::createFromWkt( featureJson.GetString("geom").c_str(),
+        nullptr, &poGeometry );
+    if( poGeometry != nullptr )
+    {
+        OGRSpatialReference *poSpatialRef =
+            poFeatureDefn->GetGeomFieldDefn( 0 )->GetSpatialRef();
+        if( poSpatialRef != nullptr)
+        {
+            poGeometry->assignSpatialReference( poSpatialRef );
+        }
+        poFeature->SetGeomFieldDirectly( 0, poGeometry );
+    }
+
+    // FIXME: Do we need extension/description and extension/attachment storing in feature?
+
+    return poFeature;
+}
+
+/*
+ * FeatureToJson()
+ */
+static CPLJSONObject FeatureToJson(OGRFeature *poFeature)
+{
+    CPLJSONObject oFeaturesJson;
+    if( poFeature == nullptr )
+    {
+        // Should not happen.
+        return oFeaturesJson;
+    }
+
+    if( poFeature->GetFID() >= 0 )
+    {
+        oFeaturesJson.Add("id", poFeature->GetFID());
+    }
+
+    OGRGeometry *poGeom = poFeature->GetGeometryRef();
+    if( poGeom )
+    {
+        char *pszWkt = nullptr;
+        if( poGeom->exportToWkt( &pszWkt ) == OGRERR_NONE )
+        {
+            oFeaturesJson.Add("geom", std::string(pszWkt));
+        }
+    }
+
+    OGRFeatureDefn *poFeatureDefn = poFeature->GetDefnRef();
+    CPLJSONObject oFieldsJson("fields", oFeaturesJson);
+    for( int iField = 0; iField < poFeatureDefn->GetFieldCount(); ++iField )
+    {
+        OGRFieldDefn *poFieldDefn = poFeatureDefn->GetFieldDefn(iField);
+        if( poFeature->IsFieldNull(iField) == TRUE )
+        {
+            oFieldsJson.AddNull("poFieldDefn->GetNameRef()");
```

Comment:
the value of the argument of AddNull() is suspicious

Reply:
Fix per 4579aeab927d70ddab6142340bf5550c6789b42e